net/peer.go

```diff
@@ -191,7 +196,7 @@ func NewPeer(
 }
 
 // Close the peer node and all its internal workers/goroutines/loops.
-func (p *Peer) Close() {
+func (p *Peer) Close() error {
```

Comment:
todo: Can we remove this return param? It appears to be unused.

If you added it because you want to implement `io.Closer`, can you please at least document this, and why it needs to implement that interface.

If there is no real technical reason for it to implement that interface, please remove it, returning errors where there are none adds a notable amount of unnecessary boilerplate/complexity and distracts readers from the important parts of the code.

Reply:
I was thinking of using the stdlib interface to keep things simple but I do agree that if we can avoid the error it makes more sense.

Reply:
Check out the last commit and let me know what you think.

Reply:
![image](https://github.com/user-attachments/assets/45e5e1fa-4bd6-4078-9f85-c217fb31adad)

Reply:
If we are only going to have two items in the array it is probably an over complication IMO and two simple props would be better.  If you think that list is likely to grow it is a nice solution :)

Reply:
My thinking was that if we want to make things a bit more indepandant and give more control to users as to when things are closed, we could have ACP, Lens, Net and HTTP all be handled by the terminator instead of hidden in different places. It will be even more useful when users want to assemble their own nodes.